examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveFramework.scala

```diff
@@ -0,0 +1,99 @@
+package com.mesosphere.usi.helloworld
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Flow
+import com.mesosphere.usi.core.Scheduler
+import com.mesosphere.usi.core.models._
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.mesos.v1.Protos.{TaskState, TaskStatus}
+
+class KeepAliveFramework(conf: Config) extends StrictLogging {
+
+  implicit val system = ActorSystem()
+  implicit val mat = ActorMaterializer()
+  implicit val ec = system.dispatcher
+
+  val client = new KeepAliveMesosClientFactory(conf).client
+
+  val runSpec = KeepAlivePodSpecHelper.runSpec
+
+  val specsSnapshot = KeepAlivePodSpecHelper.specsSnapshot(conf.getInt("keep-alive-framework.tasks-started"))
+
+  // KeepAliveWatcher looks for a terminal task and then restarts the whole pod.
+  val keepAliveWatcher: Flow[StateEvent, SpecUpdated, NotUsed] = Flow[StateEvent].mapConcat {
+    // Main state event handler. We log happy events and restart the pod if something goes wrong
+    case s: StateSnapshot =>
+      logger.info(s"Initial state snapshot: $s")
+      Nil
+
+    case PodStatusUpdated(id, Some(PodStatus(_, taskStatuses))) =>
+      import TaskState._
+      def activeTask(status: TaskStatus) = Seq(TASK_STAGING, TASK_STARTING, TASK_RUNNING).contains(status.getState)
+      // We're only interested in the bad task statuses for our pod
+      val failedTasks = taskStatuses.filterNot { case (id, status) => activeTask(status) }
```

Comment:
Nit, it would be nice to have
```suggestion
      val failedTasks = taskStatuses.filterNot(activeTask)
```

Reply:
It won't work because we have a Map here.